File: wowsinfo/src/core/model/api/Meta.ts
/// Meta should never be null so you don't need to check
class Meta {
    count?: number;
    page_total?: number;
    total?: number;
    limit?: number;
    page?: number;
    // If it's not null, it's hidden
    hidden?: object[];

    constructor(json: any) {
        this.count = json.count;
        this.page_total = json.page_total;
        this.total = json.total;
        this.limit = json.limit;
        this.page = json.page;
        this.hidden = json.hidden;
    }
}

export { Meta };

File: wowsinfo/lib/core/models/Meta.dart
/// This is the `Meta` class, shared by all API responses from Wargaming API
/// - It is usually used to determine if this account is hidden
/// - Sometimes, it can be used to see if there are more data to be downloaded (ships, consumables and more)
class Meta {
  int count;
  int pageTotal;
  int total;
  int limit;
  int page;
  /// If the account is not _hidden, it should be null
  dynamic _hidden;
  bool get isHidden => _hidden != null;

  Meta(Map<String, dynamic> json) {
    this.count = json['count'];
    this.pageTotal = json['page_total'];
    this.total = json['total'];
    this.limit = json['limit'];
    this.page = json['page'];
    this._hidden = json['_hidden'];
  }
}
